eth/tracers/native/call_flat.go

```diff
@@ -0,0 +1,297 @@
+	TransactionPosition uint64          `json:"transactionPosition"`
+	Type                string          `json:"type"`
+}
+
+type flatCallFrameMarshaling struct {
+	BlockNumber         hexutil.Uint64
+	TransactionPosition hexutil.Uint64
+}
+
+type flatCallAction struct {
+	Author         *common.Address `json:"author,omitempty"`
+	RewardType     string          `json:"rewardType,omitempty"`
+	SelfDestructed *common.Address `json:"address,omitempty"`
+	Balance        *big.Int        `json:"balance,omitempty"`
+	CallType       string          `json:"callType,omitempty"`
+	CreationMethod string          `json:"creationMethod,omitempty"`
+	From           *common.Address `json:"from,omitempty"`
+	Gas            *uint64         `json:"gas,omitempty"`
+	Init           *[]byte         `json:"init,omitempty"`
+	Input          *[]byte         `json:"input,omitempty"`
+	RefundAddress  *common.Address `json:"refundAddress,omitempty"`
+	To             *common.Address `json:"to,omitempty"`
+	Value          *big.Int        `json:"value,omitempty"`
+}
+
+type flatCallActionMarshaling struct {
+	Balance *hexutil.Big
+	Gas     *hexutil.Uint64
+	Init    *hexutil.Bytes
+	Input   *hexutil.Bytes
+	Value   *hexutil.Big
+}
+
+type flatCallResult struct {
+	Address *common.Address `json:"address,omitempty"`
+	Code    *[]byte         `json:"code,omitempty"`
+	GasUsed *uint64         `json:"gasUsed,omitempty"`
+	Output  *[]byte         `json:"output,omitempty"`
+}
+
+type flatCallResultMarshaling struct {
+	Code    *hexutil.Bytes
+	GasUsed *hexutil.Uint64
+	Output  *hexutil.Bytes
+}
+
+// flatCallTracer reports call frame information of a tx in a flat format, i.e.
+// as opposed to the nested format of `callTracer`.
+type flatCallTracer struct {
+	tracer *callTracer
+	ctx    *tracers.Context // Holds tracer context data
+	reason error            // Textual reason for the interruption
+}
+
+// newFlatCallTracer returns a new flatCallTracer.
+func newFlatCallTracer(ctx *tracers.Context, _ json.RawMessage) (tracers.Tracer, error) {
+	tracer, err := tracers.DefaultDirectory.New("callTracer", ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	t, ok := tracer.(*callTracer)
+	if !ok {
+		return nil, errors.New("internal error: embedded tracer has wrong type")
+	}
+
+	return &flatCallTracer{tracer: t, ctx: ctx}, nil
+}
+
+// CaptureStart implements the EVMLogger interface to initialize the tracing operation.
+func (t *flatCallTracer) CaptureStart(env *vm.EVM, from common.Address, to common.Address, create bool, input []byte, gas uint64, value *big.Int) {
+	t.tracer.CaptureStart(env, from, to, create, input, gas, value)
+}
+
+// CaptureEnd is called after the call finishes to finalize the tracing.
+func (t *flatCallTracer) CaptureEnd(output []byte, gasUsed uint64, err error) {
+	t.tracer.CaptureEnd(output, gasUsed, err)
+	// Parity trace considers only reports the gas used during the top call frame which doesn't include
```

Comment:
Can you please revert this too? Our gas usage is more accurate, accounting for intrinsic gas and refunds too.